deploy.sh

```diff
@@ -88,37 +150,51 @@ if [ -z $SP_APP_ID ] || [ -z $SP_APP_KEY ] || [ -z $SP_TENANT_ID ] ; then
   if ! az login -o none; then
       echo "Unable to connect to Azure" >&2
       exit 1
+  else
+      echo "Logged in to Azure"
   fi
 else
-  echo "Attempting to log in to Azure with service principal"
+  echo "Attempting to log in to Azure with service principal ${SP_APP_ID}" 
   if ! az login --service-principal -u "${SP_APP_ID}" -p "${SP_APP_KEY}" -t "${SP_TENANT_ID}"; then
     echo "Unable to connect to Azure" >&2
     exit 1
+  else
+      echo "Logged in to Azure"
   fi
 fi
 
-echo "Activating Azure subscription: ${AZURE_SUBSCRIPTION}"
-
 # Activate chosen subscription
+echo "Activating Azure subscription: ${AZURE_SUBSCRIPTION}"
 az account set -s "$AZURE_SUBSCRIPTION"
 
-echo "Checking if resource group exists: ${RESOURCE_GROUP_NAME}"
 # Create a new resource group if necessary
+echo "Checking if resource group exists: ${RESOURCE_GROUP_NAME}"
 if [[ $(az group exists --name $RESOURCE_GROUP_NAME) == false ]] ; then
   echo "Creating new resource group: ${RESOURCE_GROUP_NAME}"
   az group create -n $RESOURCE_GROUP_NAME --location $RESOURCE_GROUP_LOCATION -o table
+else
+  echo "Resource group ${RESOURCE_GROUP_NAME} found."
 fi
 
 # Create an AKS cluster
-echo "Creating AKS cluster; this may take a few monutes to complete"
+echo "Creating AKS cluster; this may take a few minutes to complete
+Resource Group: ${RESOURCE_GROUP_NAME}
+Cluster name:   ${AKS_NAME}
+Node count:     ${AKS_NODE_COUNT}
+Node VM size:   ${AKS_NODE_VM_SIZE}"
 az aks create -n $AKS_NAME -g $RESOURCE_GROUP_NAME --generate-ssh-keys --node-count $AKS_NODE_COUNT --node-vm-size $AKS_NODE_VM_SIZE -o table
 
 # Get kubectl credentials from Azure
 echo "Fetching kubectl credentials from Azure"
 az aks get-credentials -n $AKS_NAME -g $RESOURCE_GROUP_NAME -o table
 
 # Check nodes are ready
-while [[ ! x`kubectl get node | awk '{print $2}' | grep Ready | wc -l` == x${AKS_NODE_COUNT} ]] ; do echo -n $(date) ; echo " : Waiting for all cluster nodes to be ready" ; sleep 15 ; done
+nodecount="$(kubectl get node | awk '{print $2}' | grep Ready | wc -l)"
+while [[ ! x${nodecount} == x${AKS_NODE_COUNT} ]] ; do echo -n $(date) ; echo " : ${nodecount} of ${AKS_NODE_COUNT} nodes ready" ; sleep 15 ; nodecount="$(kubectl get node | awk '{print $2}' | grep Ready | wc -l)" ; done
```

Comment:
```suggestion
while [[ x${nodecount} -ne x${AKS_NODE_COUNT} ]] ; do echo -n $(date) ; echo " : ${nodecount} of ${AKS_NODE_COUNT} nodes ready" ; sleep 15 ; nodecount="$(kubectl get node | awk '{print $2}' | grep Ready | wc -l)" ; done
```

To fix the never-ending loop on MacOS